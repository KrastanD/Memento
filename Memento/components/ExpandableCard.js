import React, { useEffect, useState } from "react";
import { Card } from "react-native-elements";
import Icon from "react-native-vector-icons/FontAwesome";
import { Text, View, TouchableOpacity, Linking } from "react-native";

export default function ExpandableCard({ event }) {
  const [isOpen, setOpen] = useState(false);

  return (
    <Card>
      <View>
        <View style={{ flexDirection: "row" }}>
          <View style={{ flex: 1 }}>
            <Text style={{ fontSize: 15 }}>{event.eventTitle}</Text>
          </View>
          <View style={{ flexDirection: "column" }}>
            <Text style={{ fontSize: 15, textAlign: "right" }}>
              {getTime(event)}
            </Text>
            {!isOpen ? (
              <TouchableOpacity
                style={{ alignSelf: "flex-end" }}
                onPress={() => {
                  setOpen(!isOpen);
                }}
              >
                <Icon name="chevron-down" size={20} color="#01a699"></Icon>
              </TouchableOpacity>
            ) : (
              <TouchableOpacity
                style={{ alignSelf: "flex-end" }}
                onPress={() => {
                  setOpen(!isOpen);
                }}
              >
                <Icon name="chevron-up" size={20} color="#01a699"></Icon>
              </TouchableOpacity>
            )}
          </View>
        </View>
        {isOpen && (
          <View>
            {event.description ? (
              <Text style={{ fontSize: 14 }}>
                {"Description: " + event.description}
              </Text>
            ) : null}
            {event.location ? (
              <TouchableOpacity
                onPress={() => {
                  Linking.openURL(
                    "https://www.google.com/maps/search/?api=1&query=" +
                      event.location
                  );
                }}
              >
                <Text style={{ fontSize: 14, textDecorationLine: "underline" }}>
                  {"Location: " + event.location}
                </Text>
              </TouchableOpacity>
            ) : null}
          </View>
        )}
      </View>
    </Card>
  );
}

function getTime(event) {
  var hours = event.timePostedDate.getHours();
  var minutes = event.timePostedDate.getMinutes();
  if (minutes == "0") {
    minutes = "00";
  }
  if (hours == "0") {
    hours = "00";
  }
  return hours + ":" + minutes;
}
