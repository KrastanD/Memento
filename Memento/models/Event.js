class Event {
  constructor(json) {
    if (json == "{}") {
      this.id = "";
      this.eventTitle = "";
      this.description = "";
      this.timePostedString = "";
      this.timePostedDate = "";
    } else {
      this.id = json["_id"];
      this.eventTitle = json["eventTitle"];
      this.description = json["description"];
      var d = json["timePosted"];
      this.timePostedDate = new Date(d);
      this.timePostedString = this.timePostedDate.toLocaleDateString("en-US", {
        month: "long",
        day: "numeric",
        year: "numeric",
      });
      this.location = json["location"];
    }
  }
}
export default Event;
