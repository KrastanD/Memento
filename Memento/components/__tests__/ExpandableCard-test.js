import React from "react";
import ExpandableCard from "../ExpandableCard";
import renderer from "react-test-renderer";
import Event from "../../models/Event";

mockedJson = {
  _id: "123",
  eventTitle: "Mocked Event",
  description: "Mocked Description",
  timePosted: "Wed Apr 1 2020 13:11:13 GMT-0500 (CDT)",
};

it("snapshots expandable card component correctly", () => {
  var ev = new Event(mockedJson);
  const tree = renderer.create(<ExpandableCard event={ev} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

mockedJson2 = {
  _id: "134",
  eventTitle: "Mocking missing description",
  timePosted: "Wed Apr 1 2020 13:11:13 GMT-0500 (CDT)",
};

it("snapshots expendable card component with missing description", () => {
  var ev = new Event(mockedJson2);
  const tree = renderer.create(<ExpandableCard event={ev} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

mockedJson3 = {
  _id: "134",
  eventTitle: "Mocking with location and no description",
  timePosted: "Wed Apr 1 2020 13:11:13 GMT-0500 (CDT)",
  location: "Siebel",
};

it("snapshots expendable card component with location and without description", () => {
  var ev = new Event(mockedJson3);
  const tree = renderer.create(<ExpandableCard event={ev} />);
  expect(tree.toJSON()).toMatchSnapshot();
});

mockedJson4 = {
  _id: "134",
  eventTitle: "Mocking with location and description",
  description: "Mocked Description",
  timePosted: "Wed Apr 1 2020 13:11:13 GMT-0500 (CDT)",
  location: "Siebel",
};

it("snapshots expendable card component with location and without description", () => {
  var ev = new Event(mockedJson4);
  const tree = renderer.create(<ExpandableCard event={ev} />);
  expect(tree.toJSON()).toMatchSnapshot();
});
