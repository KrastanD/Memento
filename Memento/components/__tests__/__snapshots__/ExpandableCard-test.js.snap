// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots expandable card component correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 0,
      "padding": 15,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
              }
            }
          >
            Mocked Event
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
                "textAlign": "right",
              }
            }
          >
            13:11
          </Text>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "flex-end",
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`snapshots expendable card component with location and without description 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 0,
      "padding": 15,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
              }
            }
          >
            Mocking with location and no description
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
                "textAlign": "right",
              }
            }
          >
            13:11
          </Text>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "flex-end",
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`snapshots expendable card component with location and without description 2`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 0,
      "padding": 15,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
              }
            }
          >
            Mocking with location and description
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
                "textAlign": "right",
              }
            }
          >
            13:11
          </Text>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "flex-end",
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`snapshots expendable card component with missing description 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderColor": "#e1e8ee",
      "borderWidth": 1,
      "margin": 15,
      "marginBottom": 0,
      "padding": 15,
      "shadowColor": "rgba(0,0,0, .2)",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 1,
      "shadowRadius": 1,
    }
  }
>
  <View
    style={
      Object {
        "backgroundColor": "transparent",
      }
    }
  >
    <View>
      <View
        style={
          Object {
            "flexDirection": "row",
          }
        }
      >
        <View
          style={
            Object {
              "flex": 1,
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
              }
            }
          >
            Mocking missing description
          </Text>
        </View>
        <View
          style={
            Object {
              "flexDirection": "column",
            }
          }
        >
          <Text
            style={
              Object {
                "fontSize": 15,
                "textAlign": "right",
              }
            }
          >
            13:11
          </Text>
          <View
            accessible={true}
            focusable={true}
            isTVSelectable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignSelf": "flex-end",
                "opacity": 1,
              }
            }
          >
            <Text />
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
